int8 x_angle
int8 y_angle
int8 z_angle